crates/chain/src/persist.rs

```diff
@@ -1,6 +1,26 @@
-use core::convert::Infallible;
-
 use crate::Append;
+use core::fmt;
+
+#[derive(Debug, Clone)]
+/// Errors returned by [`PersistBackend`] when staging and committing changes.
```

Comment:
🔧 This description doesn't capture that this can also be thrown for load errors. How about something simple like:
```suggestion
/// Errors returned by [`PersistBackend`] when writing or loading data.
```

Reply:
Agreed. I am also working on more expressive error variants by wrapping the errors.